testing work flow for merging recommender repo

russaa opened the pull request:
Pull Request for testing the work flow for merging from the repository where recommender-functionality is implemented

This PR deliberately contains outdated code that can not be automatically merged, in order to test change requests & updates to the PR etc.

TODO for this pull request
 * [x] notify upstream maintainer w.r.t. pending PR
 * [x] merge upstream changes into recommender repo & update PR, so that automatic merging is possible
 * [x] test commenting & request changes to PR
 
**NOTE** the target branch (upstream) will be deleted after this test is finished

TabeaOt committed:
migrated changes from https://github.com/mmig/ki-campus-bot

russaa committed:
BUGFIX removed duplication when referencing inform_coursesearch slot

must set level-value via /inform_coursesearch{"level":"Experte"}'} not /inform_coursesearch/inform_coursesearch{"level":"Experte"}'}

russaa committed:
FIX indent format: consistently use tabs for indenting code

russaa committed:
BUGFIX [reapply fix] removed duplication when referencing inform_coursesearch slot

must set level-value via /inform_coursesearch{"level":"Experte"}'} not /inform_coursesearch/inform_coursesearch{"level":"Experte"}'}

russaa committed:
BUGFIX for typo fomat -> format

russaa committed:
BUGFIX use correct method name levels_db() instead of duplicate duration_db()

russaa committed:
updated usage-instructions for development

russaa committed:
REVERT previous commit 76b7c1e02d071a740164b3800d3b7183f5371500

russaa committed:
Merge branch 'main' of https://github.com/mmig/chatbot-kim

russaa committed:
REVERT previous commit 76b7c1e02d071a740164b3800d3b7183f5371500

russaa committed:
updated usage-instructions for development

russaa committed:
implemented querying the recommender endpoint & showing its results

russaa committed:
added instructions for configuring the recommender endpoint

russaa committed:
git-ignore configuration/credentials file for recommender endpoint

russaa committed:
added utterances in case of error-results when querying course-recommendations

russaa committed:
FIX use absolute path for loading config-file instead of relative path w.r.t. current-working-directory

russaa committed:
added some DEBUG-output for get-recommendations-action

TabeaOt committed:
Merge pull request #1 from mmig/feature/recommender_integration

Feature/recommender integration

TabeaOt committed:
small changes texts, action delete slot value

russaa committed:
REFACTOR extracted recommender-specific actions to separate file

russaa committed:
BUGFIX avoid duplicate class-names for actions

russaa committed:
REFACTOR use own low-resources implementation for loading YAML file/configuration

TabeaOt committed:
changed training data, updated action, config

russaa committed:
FIX update docker configuration for new config-loading mechanism

 * install dependency for YAML library in docker for action server
 * do load/mount recommender-config file via docker-compose config when running action server

russaa committed:
FIX for spacy-pipeline: do install German spacy model via docker configuration

russaa committed:
FIX indentation for volume configuration

russaa committed:
added separate endpoints-configuration for docker

 * added specific endpoints-configuration file for docker to allow using default endpoints-configuration for local rasa shell

russaa committed:
FEATURE implemented loading action's response-texts from YAML file

 * implemented helpers for loading YAML file with response-texts (i.e. "utterances") that are used within action implementations
 * extracted response-texts (i.e. "utterances") from actions/actions.py to action_responses/actions_responses.yml
 * added action_responses/ as mounted volume in docker-compose configuration

russaa committed:
Merge branch 'main' into refactor/actions

# Conflicts:
#	rasa/KI-Campus_de/actions/actions.py

russaa committed:
added (placeholder) label for course-title

 + use relative import for responses.py
 + removed commented-out utterances

russaa committed:
extracted response-texts/utterances from actions_recommender.py

russaa committed:
FIX explicitly load YAML files with UTF-8 encoding

russaa committed:
WORKAROUND for BUG in socket.io adapter: do not dispatch utterances with empty text (i.e. avoid buttons-only utterances)

NOTE currently socket.io adapter will cause error on encountering messages with empty/None text, e.g. messages that only contain a button group
WORKAROUND do include some (non-empty) text when dispatching button-groups within the same message

russaa committed:
IMPROVE only query course_title once per action invocation from responses

russaa committed:
REFACTOR extract/use response-strings for course-titles

russaa committed:
IMPROVE only query course_title once per action invocation from responses (2)

russaa committed:
NOOP

re-formatted to match formatting in
rasa/KI-Campus_en/actions/actions.py

russaa committed:
extracted English response texts

russaa committed:
adapted docker for English chatbot for response-text loading

russaa committed:
FIX disabled installing spacy model since NLU pipeline was reverted to not use spacy NLP

russaa committed:
FIX instructions for starting action server to ensure that all actions are started

i.e. also actions that are located in other python-files under actions/*.py, other than actions.py

russaa committed:
IMPROVE documentation of parameters in response-texts

theresazobel committed:
Add first version of human handoff (#6)

theresazobel committed:
Update to Rasa 3.1 (#7)

theresazobel committed:
Tz/handle insults (#8)

* Add list of german instults and handle them when used

* Merge main

* Add first list of english insults and how to handle them

* Rename to KIM

theresazobel committed:
Set current course title to None after receiving the achievements (#9)

theresazobel committed:
Update docker files for new rasa version (#10)

theresazobel committed:
Tz/cleanup v2 (#11)

* Clean up files, fix intendations, fix spelling, fix end of file newline

* Update faqs for englisch bot, cleanup and enhance faqs for german bot

TabeaOt committed:
implemented new slot type

TabeaOt committed:
Merge remote-tracking branch upstream/main to main

theresazobel commented:
Hello, danke für euren MR. Mir ist beim ersten Durchsehen aufgefallen, dass wir doch einige intents doppelt haben. Das müssen wir noch lösen, da es sonst definitiv Probleme gibt. Beispiele: 
- course costs
- affirm
- hello
- frage zu themen
- get achievements
- intent faq (ihr habt hier ein eigenes intent, bei mir ist FAQ eine gesamte Liste an verschiedenen Fragen)

Weiters ist mir aufgefallen, dass ihr prüft ob der Nutzer eingeloggt ist. Soweit ich weiß, ist der Chatbot eh nur verfügbar, wenn der Nutzer eingeloggt ist (das wird sich sicher mal in der Zukunft ändern). 

Lasst uns gerne demnächst mal über den MR telefonieren und die Punkte besprechen.

theresazobel committed:
Cleanup, fixing and updating some utterances (#12)

TabeaOt committed:
update new slot type banch w main

TabeaOt committed:
small correction domain

TabeaOt committed:
merge upstream into new_slot_type branch

russaa committed:
Merge branch 'main' into refactor/actions

russaa committed:
implemented login check & querying user profile & using profile data for recommendations

russaa committed:
REVERT removed activation/configuration for socketio

TabeaOt committed:
Merge pull request #2 from mmig/refactor/actions

REFACTOR actions

theresazobel committed:
Set current course to NONE when chatbot did not understand the course… (#13)

* Set current course to NONE when chatbot did not understand the course correctly

* Add EOL

theresazobel committed:
Add first simple feedback loop (#14)

theresazobel committed:
Cleanup (#15)

theresazobel committed:
Disable buttons for now, xikolo needs to be fixed first (#16)

theresazobel committed:
Fix selection of current course by user (#17)

theresazobel committed:
Fix spelling mistake in action (#18)

* Fix spelling mistake in action

* Fix spelling mistake in action

TabeaOt committed:
small updates stories

russaa committed:
BUGFIX initialize variable "response" for all cases / code-paths

russaa committed:
FIX handle empty/None values for slots in actions

russaa committed:
FIX handle empty/None values for slots in validate-action

russaa committed:
added line-break after debug-output & marked all print-debug instructions with DEV comment FIXME DEBUG

russaa committed:
NOOP (removed unused import)

russaa committed:
FIX WORKAROUND to ensure that the message-text precedes button-list

TabeaOt committed:
workaround slot validation

TabeaOt committed:
Merge branch 'main' of https://github.com/mmig/chatbot-kim into main

theresazobel committed:
Add more helpful yes and no buttons for users to click (#19)

* Add more helpful yes and no buttons for users to click

* EOF

russaa committed:
Merge branch 'main' of https://github.com/KI-Campus/chatbot-kim into merge_upstream/from_upstream_main

Resolved Conflicts:
	rasa/KI-Campus_de/GreetBot_de/domain.yml
	rasa/KI-Campus_de/actions/actions.py
	rasa/KI-Campus_en/actions/actions.py

russaa committed:
Merge pull request #3 from mmig/merge_upstream/from_upstream_main

Merge upstream/from upstream main

russaa committed:
HACK fix removed/missing intent utter_end_of_conversation

temporary fix missing intent that was removed from
domain.yml
-> old version for intent copied to
DFKIBot_de/domain.yml

russaa committed:
Merge pull request #4 from mmig/bugfix/readd_missing_upstream_action

HACK fix removed/missing intent utter_end_of_conversation

russaa committed:
REFACTOR extract language code for achievements-request from implementation file to labels file

russaa committed:
Merge pull request #5 from mmig/refactor/actions

Refactor/actions extract language code to labels file

TabeaOt committed:
Merge branch 'main' of https://github.com/mmig/chatbot-kim into main

TabeaOt committed:
Merge branch 'main' of https://github.com/mmig/chatbot-kim into main

TabeaOt committed:
deleted end_of_conversation,

TabeaOt committed:
KIC-Bot changed intent "inform" to "courses"

TabeaOt committed:
rename coursesearch "recommender", update stories

TabeaOt committed:
quick fix responses, cleanup

TabeaOt committed:
response fix

russaa commented:
using "cleaner" PR https://github.com/KI-Campus/chatbot-kim/pull/21 instead
